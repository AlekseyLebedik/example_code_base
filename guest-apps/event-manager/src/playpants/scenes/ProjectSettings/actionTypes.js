const prefix = 'playpants/PROJECT_SETTINGS';

export const FETCH_AVAILABLE_USERS = `${prefix}.AVAILABLE_USERS`;
export const SEARCH_USERS = `${prefix}.SEARCH_USERS`;
export const FILTER_GROUPS = `${prefix}.FILTER_GROUPS`;
export const CHANGE_TAB = `${prefix}.CHANGE_TAB`;

export const UPDATE_PROJECT_SETTING = `${prefix}.UPDATE_PROJECT_SETTING`;
export const UPDATE_PROJECT_SETTING_SUCCEED = `${prefix}.UPDATE_PROJECT_SETTING_SUCCEED`;

export const FETCH_PROJECT_SCHEMA = `${prefix}.FETCH_PROJECT_SCHEMA`;
export const FETCH_PROJECT_SCHEMA_SUCCEED = `${prefix}.FETCH_PROJECT_SCHEMA_SUCCEED`;
export const CLEAR_PROJECT_SCHEMA = `${prefix}.CLEAR_PROJECT_SCHEMA`;
