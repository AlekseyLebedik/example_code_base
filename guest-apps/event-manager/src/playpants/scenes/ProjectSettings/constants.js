export const PROJECT_SETTINGS_LS_KEY = 'project_settings_history';
export const SETTINGS_TABS = {
  responsibilities: 'Responsibilities',
  'gamertag-management': 'Groups',
  settings: 'Settings',
};

export const RESPONSIBILITIES_SUB_TABS = {
  groups: 'Groups',
  users: 'Users',
};

export const SETTINGS_SUB_TABS = {
  global: 'Global',
  events: 'Events',
  activities: 'Activities',
  'name-mapping': 'Name Mapping',
  notifications: 'Notifications',
  colors: 'Colors',
};

export const PROJECT_ENVIRONMENTS = {
  cert: 'Certification',
  dev: 'Development',
  live: 'Live',
};

export const ICONS = [
  'android',
  'iphone',
  'psp',
  'xbox',
  'pc-steam',
  'wii-u',
  'wii',
  'pc-bnet',
  'default',
];

export const GROUPS = ['All', 'Demonware', 'Treyarch', 'Infinity Ward'];

export const PROJECT_SETTINGS_URL = 'project-settings';

export const RESP_URL = `${PROJECT_SETTINGS_URL}/responsibilities`;

export const RESP_REDIRECT_URL = `${RESP_URL}/groups`;

export const GAMERTAGS_URL = `${PROJECT_SETTINGS_URL}/gamertag-management`;

export const GAMERTAGS_REDIRECT_URL = `${GAMERTAGS_URL}/groups`;

export const SETTINGS_URL = `${PROJECT_SETTINGS_URL}/settings`;

export const SETTINGS_REDIRECT_URL = `${SETTINGS_URL}/global`;
