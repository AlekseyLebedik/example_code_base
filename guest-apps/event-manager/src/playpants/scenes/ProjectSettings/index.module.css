.bodyContainer {
  overflow: scroll;
  padding: 20px;
  height: 100%;
}

.tabPane {
  margin-top: 25px;
  display: flex;
  flex-direction: row;
  align-content: stretch;
  overflow: hidden;
}

.sidebar {
  flex-direction: column;
  background-color: #f5f5f6;
  flex-flow: column nowrap;
  overflow: hidden;
  display: flex;
  width: 210px;
  min-width: 210px;
  height: 75vh;
}

.authBody {
  overflow-x: auto;
  margin-left: 20px;
}

.projectSettingsContentHeader {
  padding: 7px 0px 30px 10px;
}

.toastContainer {
  background-color: #e5e5e6;
  color: rgba(0, 0, 0, 0.75);
}

@media only screen and (max-width: 768px) {
  .tabPane {
    display: block;
  }

  .sidebar {
    margin-bottom: 20px;
    width: 100%;
  }

  .authBody {
    margin-left: 0;
  }
}
