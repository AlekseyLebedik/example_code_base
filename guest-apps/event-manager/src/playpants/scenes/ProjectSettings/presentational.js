import React from 'react';
import PropTypes from 'prop-types';

import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';

import AdapterLink from '@demonware/devzone-core/components/RouteAdapterLink';
import RedirectHandler from 'playpants/components/RedirectHandler';
import {
  SETTINGS_TABS,
  PROJECT_SETTINGS_URL,
  RESP_REDIRECT_URL,
} from './constants';

import Responsibilities from './components/Responsibilities';
import EventManagerSettings from './components/EventManagerSettings';
import GamertagManagement from './components/GamertagManagement';

import { useStyles } from './theme';

const StatelessProjectSettings = props => {
  const classes = useStyles();
  const { baseUrl, hasCurrentProjectSettings, match } = props;
  const { category: selectedTab, subcategory: selectedSubTab } = match.params;
  return (
    <RedirectHandler
      matchParam={selectedTab}
      list={Object.keys(SETTINGS_TABS)}
      url={`${baseUrl}${RESP_REDIRECT_URL}`}
    >
      <>
        <Tabs value={selectedTab} indicatorColor="primary" textColor="primary">
          {Object.entries(SETTINGS_TABS)
            .filter(([t]) => hasCurrentProjectSettings || t === 'settings')
            .map(([key, label]) => (
              <Tab
                key={key}
                value={key}
                label={label}
                component={AdapterLink}
                to={`${baseUrl}${PROJECT_SETTINGS_URL}/${key}`}
              />
            ))}
        </Tabs>
        {selectedTab === 'responsibilities' && hasCurrentProjectSettings && (
          <Responsibilities
            {...props}
            classes={classes}
            selectedTab={selectedSubTab}
          />
        )}
        {selectedTab === 'gamertag-management' && hasCurrentProjectSettings && (
          <GamertagManagement
            {...props}
            classes={classes}
            selectedTab={selectedSubTab}
          />
        )}
        {selectedTab === 'settings' && (
          <EventManagerSettings
            {...props}
            classes={classes}
            selectedTab={selectedSubTab}
          />
        )}
      </>
    </RedirectHandler>
  );
};

StatelessProjectSettings.propTypes = {
  baseUrl: PropTypes.string.isRequired,
  hasCurrentProjectSettings: PropTypes.bool.isRequired,
  match: PropTypes.object.isRequired,
  selectedTab: PropTypes.string,
};

StatelessProjectSettings.defaultProps = {
  selectedTab: null,
};

export default StatelessProjectSettings;
