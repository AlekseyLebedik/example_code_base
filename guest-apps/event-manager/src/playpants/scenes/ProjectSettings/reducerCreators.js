import { INITIAL_STATE } from '@demonware/devzone-core/helpers/reducers';

export const updateReducer =
  (prefix, append) =>
  (state = INITIAL_STATE, action) => {
    switch (action.type) {
      case `${prefix}`:
        return {
          ...state,
          loading: true,
        };
      case `${prefix}_SUCCESS`: {
        return {
          ...state,
          data: append ? [...state.data, ...action.response] : action.response,
          loading: false,
        };
      }
      case `${prefix}_FAILED`: {
        return {
          ...state,
          error: action.error,
          loading: false,
        };
      }
      default:
        return {
          ...state,
        };
    }
  };
