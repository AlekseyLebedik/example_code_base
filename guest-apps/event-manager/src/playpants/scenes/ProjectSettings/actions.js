import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import { createFetch } from '@demonware/devzone-core/helpers/actions';
import * as creators from 'playpants/components/App/actionCreators';
import * as AT from './actionTypes';

export const fetchAvailableUsers = creators.fetchAvailableUsers(
  AT.FETCH_AVAILABLE_USERS
);

export const searchUsers = query => ({
  type: AT.SEARCH_USERS,
  query,
});

export const filterGroups = group => ({
  type: AT.FILTER_GROUPS,
  group,
});

export const changeTab = (tabType, selectedTab) => ({
  type: AT.CHANGE_TAB,
  tabType,
  selectedTab,
});

export const updateProjectSetting = (projectID, setting, data) => ({
  type: AT.UPDATE_PROJECT_SETTING,
  projectID,
  setting,
  data,
});

export const updateProjectSettingSucceed = settings => ({
  type: AT.UPDATE_PROJECT_SETTING_SUCCEED,
  settings,
});

export const fetchProjectSchema = schemaID =>
  createFetch(AT.FETCH_PROJECT_SCHEMA, null, { schemaID });

export const fetchProjectSchemaSucceed = schema => ({
  type: AT.FETCH_PROJECT_SCHEMA_SUCCEED,
  schema,
});

export const clearProjectSchema = () => ({
  type: AT.CLEAR_PROJECT_SCHEMA,
});

export const fetchFailed = error => dispatch =>
  dispatch(nonCriticalHTTPError(error));
