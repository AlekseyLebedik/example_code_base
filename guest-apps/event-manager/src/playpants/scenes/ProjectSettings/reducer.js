import reduceReducers from 'reduce-reducers';
import { combineReducers } from 'redux';
import {
  createFetchReducer,
  INITIAL_STATE as FETCH_INITIAL_STATE,
} from '@demonware/devzone-core/helpers/reducers';
import { parseSchema } from 'playpants/components/App/helpers';
import Responsibilities from './components/Responsibilities/reducer';
import * as AT from './actionTypes';

const availableUsersReducer = createFetchReducer(AT.FETCH_AVAILABLE_USERS);
const userListFiltersReducer = (state = { query: '' }) => state;

const eventsReducer = (state = FETCH_INITIAL_STATE, action) => {
  switch (action.type) {
    case `${AT.FETCH_PROJECT_SCHEMA}_FETCH_SUCCESS`:
      return {
        ...state,
        data: parseSchema(action.data),
        next: action.next,
        nextPageToken: action.nextPageToken,
      };
    case AT.CLEAR_PROJECT_SCHEMA:
      return {
        ...state,
        data: {},
      };
    default:
      return {
        ...state,
      };
  }
};

const usersReducer = combineReducers({
  availableUsers: availableUsersReducer,
  userListFilters: userListFiltersReducer,
});

export default combineReducers({
  Responsibilities,
  Schemas: reduceReducers(
    FETCH_INITIAL_STATE,
    createFetchReducer(AT.FETCH_PROJECT_SCHEMA),
    eventsReducer
  ),
  Users: usersReducer,
});
