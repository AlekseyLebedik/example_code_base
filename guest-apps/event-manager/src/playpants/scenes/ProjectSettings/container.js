import React, { useEffect } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';

import {
  activitySettingsSelector,
  getBaseURL,
  hasCurrentProjectSettingsSelector,
  projectSettingsIdSelector,
} from 'playpants/components/App/selectors';
import { useEMPermissionsResult } from 'playpants/hooks';
import * as actions from './actions';

import StatelessProjectSettings from './presentational';

export const ProjectSettings = props => {
  const permissions = useEMPermissionsResult();

  useEffect(() => {
    props.onFetchAvailableUsers();
  }, []);

  return <StatelessProjectSettings {...props} permissions={permissions} />;
};

ProjectSettings.propTypes = {
  onFetchAvailableUsers: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const { history, match } = ownProps;
  return {
    activitySettings: activitySettingsSelector(state),
    baseUrl: getBaseURL(state),
    hasCurrentProjectSettings: hasCurrentProjectSettingsSelector(state),
    history,
    match,
    projectSettingsID: projectSettingsIdSelector(state),
  };
};

const mapDispatchToProps = dispatch => ({
  addUserToast: message =>
    dispatch(GlobalSnackBarActions.show(message, 'success')),
  onFetchAvailableUsers: bindActionCreators(
    actions.fetchAvailableUsers,
    dispatch
  ),
  updateProjectSetting: bindActionCreators(
    actions.updateProjectSetting,
    dispatch
  ),
});

const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  ...ownProps,
  ...stateProps,
  ...dispatchProps,
  updateProjectSetting: (setting, data) =>
    dispatchProps.updateProjectSetting(
      stateProps.projectSettingsID,
      setting,
      data
    ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
  mergeProps
)(ProjectSettings);
