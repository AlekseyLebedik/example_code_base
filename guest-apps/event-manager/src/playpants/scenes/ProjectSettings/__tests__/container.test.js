import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';

import mockState from 'playpants/testUtils/mockState';
import { projectSettingsProps as props } from 'playpants/testUtils/projectSettingsProps';
import ProjectSettings from '../index';

describe('ProjectSettings', () => {
  const store = configureMockStore()(mockState);
  const wrapper = renderer.create(
    <Provider store={store}>
      <Router>
        <ProjectSettings {...props} />
      </Router>
    </Provider>
  );

  it('renders default values', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
