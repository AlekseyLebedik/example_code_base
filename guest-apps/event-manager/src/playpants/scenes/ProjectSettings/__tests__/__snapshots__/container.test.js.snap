// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSettings renders default values 1`] = `null`;
